tests/cloner_test.go

```diff
@@ -0,0 +1,133 @@
+
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	"kubevirt.io/containerized-data-importer/pkg/controller"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+)
+
+const (
+	testSuiteName        = "Cloner Test Suite"
+	namespacePrefix      = "cloner"
+	sourcePodFillerName  = "fill-source"
+	sourcePVCName        = "source-pvc"
+	fillData             = "123456789012345678901234567890123456789012345678901234567890"
+	testFile             = utils.DefaultPvcMountPath + "/source.txt"
+	fillCommand          = "echo \"" + fillData + "\" >> " + testFile
+	cloneCompleteTimeout = 10 * time.Second
+)
+
+var _ = Describe(testSuiteName, func() {
+	f, err := framework.NewFramework(namespacePrefix, &framework.Config{})
+	if err != nil {
+		Fail("Unable to create framework struct")
+	}
+
+	var sourcePvc *v1.PersistentVolumeClaim
+
+	BeforeEach(func() {
+		// Create the source PVC and populate it with a file, so we can verify the clone.
+		sourcePvc, err = f.CreatePVCFromDefinition(utils.NewPVCDefinition(sourcePVCName, "1G", nil, nil))
+		Expect(err).ToNot(HaveOccurred())
+		pod, err := f.CreatePod(utils.NewPodWithPVC(sourcePodFillerName, fillCommand, sourcePvc))
+		Expect(err).ToNot(HaveOccurred())
+		err = f.WaitTimeoutForPodStatus(pod.Name, v1.PodSucceeded, utils.PodWaitForTime)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if sourcePvc != nil {
+			err = f.DeletePVC(sourcePvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	It("Cloner should clone data from PVC in name space to other PVC in same name space", func() {
+		targetPvc, err := f.CreatePVCFromDefinition(utils.NewPVCDefinition(
+			"target-pvc",
+			"1G",
+			map[string]string{controller.AnnCloneRequest: f.Namespace.Name + "/" + sourcePVCName},
+			nil))
+		Expect(err).ToNot(HaveOccurred())
+		err = f.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, targetPvc.Name)
+		Expect(err).ToNot(HaveOccurred())
+
+		sourcePod, err := f.FindPodByPrefix(common.CLONER_SOURCE_PODNAME)
+		Expect(err).ToNot(HaveOccurred())
+		targetPod, err := f.FindPodByPrefix(common.CLONER_TARGET_PODNAME)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(strings.Compare(sourcePod.Spec.NodeName, targetPod.Spec.NodeName) == 0).To(BeTrue())
+
+		err = f.WaitTimeoutForPodStatus(sourcePod.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		err = f.WaitTimeoutForPodStatus(targetPod.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		// Clone is completed, verify the content matches the source.
+		Expect(verifyTargetContent(f, f.Namespace, targetPvc)).To(BeTrue())
+		// Clean up PVC, the AfterEach will also clean it up, through the Namespace delete.
+		if targetPvc != nil {
+			err = f.DeletePVC(targetPvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	It("Cloner should clone data from PVC in name space to other PVC in different name space", func() {
```

Comment:
More concise:
`It("should clone data across different namespaces", ...`

Reply:
done